exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/vendors/Default.kt

```diff
@@ -1066,14 +1066,18 @@ abstract class VendorDialect(
     }
 
     fun filterCondition(index: Index): String? {
-        return if (currentDialect is PostgreSQLDialect) {
-            index.filterCondition?.let {
-                QueryBuilder(false)
-                    .append(" WHERE ").append(it)
-                    .toString()
+        return when (currentDialect) {
+            is PostgreSQLDialect, is SQLServerDialect, is SQLiteDialect -> {
+                index.filterCondition?.let {
+                    QueryBuilder(false)
+                        .append(" WHERE ").append(it)
+                        .toString()
+                }
+            }
+            else -> {
+                exposedLogger.warn("Index creation with a filter condition is not supported in ${currentDialect.name}")
+                null
```

Comment:
This behavior is partially consistent with other `createIndex()` rules ([example 1](https://github.com/JetBrains/Exposed/blob/main/exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/vendors/H2.kt#L229), [example 2](https://github.com/JetBrains/Exposed/blob/main/exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/vendors/SQLiteDialect.kt#L219)), in that a warning is logged, but then usually the entire index is just not created. In this case, if a vendor doesn't support partial index, the provided condition will just be ignored and a full index created instead.

If anybody believes that the index shouldn't be created at all for consistency, I can adjust this by refactoring the default `createIndex()` a little (maybe an early return).

Reply:
Just wondering, why did we decide to go ahead and create a full index in this case instead? Also, if we stick to that, it would be a good idea to add it to the KDoc.